elements/map/src/plugins/globe/openglobus.js

```diff
@@ -0,0 +1,359 @@
+import {
+  Globe as OgGlobe,
+  LonLat,
+  Vec3,
+  CanvasTiles,
+  XYZ,
+  WMS,
+  OpenStreetMap,
+  Vector,
+  Entity,
+} from "@openglobus/og";
+import { distributeTileToIdealMap } from "./methods.js";
+import { transform } from "../../helpers/index.js";
+import OSM from "ol/source/OSM.js";
+import XYZ_ol from "ol/source/XYZ.js";
+import WMS_ol from "ol/source/TileWMS.js";
+import Vector_ol from "ol/source/Vector.js";
+import { getLayerById } from "../../helpers/layer.js";
+
+/** @typedef {import("./types").ExtendedOLLayer} ExtendedOLLayer */
+
+let globus;
+
+export const createGlobe = ({ EOxMap, target, mapPool }) => {
+  globus = new OgGlobe({
+    target,
+    layers: EOxMap.getFlatLayersArray(EOxMap.map.getLayers().getArray())
+      .map((l) => createGlobusLayer(l, mapPool))
+      .filter((l) => l !== undefined),
+    sun: { active: false },
+    atmosphereEnabled: false,
+  });
+
+  const height = 21050000 / Math.pow(2, EOxMap.map.getView().getZoom() - 1);
+  const center = EOxMap.map.getView().getCenter();
+  const newCenter = transform(
+    center,
+    EOxMap.map.getView().getProjection().getCode(),
+    "EPSG:4326",
+  );
+  globus.planet.camera.setLonLat(
+    new LonLat(newCenter[0], newCenter[1], height),
+    new LonLat(newCenter[0], newCenter[1], 0),
+    Vec3.NORTH,
+  );
+
+  const style = document.createElement("style");
+  style.textContent = `
+    .og-inner {
+      height: 100%;
+    }
+    .og-inner *:not(canvas) {
+      display: none !important;
+    }
+  `;
+  target.appendChild(style);
+  return globus;
+};
+
+export const refreshGlobe = () => {
+  if (globus) {
+    globus.planet.layers.forEach((l) => {
+      if (l instanceof CanvasTiles) {
+        l.abortLoading();
+        // Force redraw of all tiles by temporarily setting animated flag
+        l.animated = true;
+        setTimeout(() => (l.animated = false), 200);
+      }
+    });
+  }
+};
+
+export const createGlobusLayer = (olLayer, mapPool) => {
+  const source = olLayer.getSource && olLayer.getSource();
+  const id = olLayer.get("id");
+  if (olLayer.get("type") === "Group") {
+    return undefined;
+  }
+  if (source instanceof OSM) {
+    return new OpenStreetMap(id, {
+      isBaseLayer: olLayer.get("base"),
+      visibility: olLayer.getVisible(),
+      opacity: olLayer.getOpacity(),
+    });
+  }
+
+  if (source instanceof XYZ_ol) {
+    return new XYZ(id, {
+      isBaseLayer: olLayer.get("base"),
```

Comment:
what is the "base" property of the ol layer?

I'm not very confident in what `isBaseLayer` of og actually does - all I could figure out is that the Layer Control of og takes this flag into account, and that there should be only 1 base layer on the planet... there are also some minor texture things. Not sure if we should set this property at all tbh